src/components/updated-terms-conds/updated-terms-conds-controller.ts

```diff
@@ -1,52 +1,81 @@
-      sessionId,
-      clientSessionId
-    );
-    req.session.redirectUri = clientInfo.redirectUri;
+    return async function (req: Request, res: Response) {
+        const clientInfo = await callClientInfo(req, res, service);
+        req.session.redirectUri = clientInfo.redirectUri;
+        req.session.state = clientInfo.state;
+
+        res.render("updated-terms-conds/index.njk");
+    };
+}
+
+export function updatedTermsCondsMandatoryGet(
+    service: UpdateTermsAndCondsServiceInterface = updateTermsAndCondsService()
+): ExpressRouteFunc {
+    return async function (req: Request, res: Response) {
+        const clientInfo = await callClientInfo(req, res, service);
+        req.session.redirectUri = clientInfo.redirectUri;
+        req.session.state = clientInfo.state;
+
+        const view = clientInfo.serviceType === "mandatory" ? "index-mandatory.njk" : "index.njk";
 
-    res.render("updated-terms-conds/index.njk");
-  };
+        return res.render("updated-terms-conds/" + view);
+    };
+}
+
+export function updatedTermsCondsOptionalGet(
+    service: UpdateTermsAndCondsServiceInterface = updateTermsAndCondsService()
+): ExpressRouteFunc {
+    return async function (req: Request, res: Response) {
+        const clientInfo = await callClientInfo(req, res, service);
+        req.session.redirectUri = clientInfo.redirectUri;
+        req.session.state = clientInfo.state;
+
+        const view = clientInfo.serviceType === "optional" ? "index-optional.njk" : "index.njk";
+
+        return res.render("updated-terms-conds/" + view);
+    };
 }
 
 export function updatedTermsCondsPost(
-  service: UpdateTermsAndCondsServiceInterface = updateTermsAndCondsService()
+    service: UpdateTermsAndCondsServiceInterface = updateTermsAndCondsService()
 ): ExpressRouteFunc {
-  return async function (req: Request, res: Response) {
-    const updatedTermsAndCondsValue = true;
-    const { email } = req.session.user;
-    const sessionId = res.locals.sessionId;
-    const redirectUri = req.session.redirectUri;
-    const acceptOrReject = req.body.acceptOrReject;
-
-    if (acceptOrReject === "reject") {
-      return res.redirect(
-        redirectUri + "?error_code=rejectedTermsAndConditions"
-      );
-    }
-
-    if (acceptOrReject === "accept") {
-      if (
-        await service.updateProfile(sessionId, email, updatedTermsAndCondsValue)
-      ) {
-        res.redirect(PATH_NAMES.AUTH_CODE);
-      } else {
-        throw new Error("Unable to update user profile");
-      }
-    }
-  };
+    return async function (req: Request, res: Response) {
+        const updatedTermsAndCondsValue = true;
+        const {email} = req.session.user;
+        const sessionId = res.locals.sessionId;
+        const redirectUri = req.session.redirectUri;
+        const state = req.session.state;
+        const acceptOrReject = req.body.acceptOrReject;
+
+        if (acceptOrReject === "reject") {
```

Comment:
you only want to add state is the session has something this would add an empty param?

Reply:
The state gets set on the session from the calls to the clientInfo lambda on any of the 3 get requests, so I don't think that field will be empty.